### PR TITLE
feat: MLSolver hashe test and compilation in the CI

### PR DESCRIPTION
**Pull request by Corentin-Allaire**
This MR adds the Ml Ambiguity solver to the CI. For now, only hashes tests are available, but once the performance evaluation for the solver has been refactored to use root, we should also add it to the test.

### CODE REVIEW

File: cmake/ActsConfig.cmake.in [hunk 1]
```diff
@@ -84,9 +84,6 @@ endif()
 
 
 # dependencies which cannot be searched in CONFIG mode
-if(PluginOnnx IN_LIST Acts_COMPONENTS)
-  find_package(OnnxRuntime REQUIRED)
-endif()
```
Comment (andiwand) [inline cmake/ActsConfig.cmake.in:89]: why is this necessary?

Comment (Corentin-Allaire) [reply cmake/ActsConfig.cmake.in:89]: You mean why is it necessary to remove this ? If you don't then when compiling against ACTS (downstream CI) we include Onnx twice. If you ask why this was here to begin with I am not sure. @paulgessinger mention we need to remplace this by a `find_dependency` so maybe he can comments about this.

Comment (Corentin-Allaire) [reply cmake/ActsConfig.cmake.in:89]: Yes I am not sure what is happening. The error I was seeing without removing those lines was that `find_package(OnnxRuntime REQUIRED)` was called twice. But I am not sure why

Comment (andiwand) [reply cmake/ActsConfig.cmake.in:89]: to me it looks like that `find_package` will export a target that is already exported by `autodiff` which is generically called `Core`. as far as I understand `find_dependency` would be the correct thing in `*Config.cmake`. `find_dependency` does not work because there are no config files for onnx.

I dont know what would be the correct thing to do here but since removing the `find_package` solves it I would say it is good enough for now

**Review by andiwand — APPROVED** [review]
